### DISCUSSION TITLE
Unable To Remove Assigned IP After Deauthorize and Delete A Member Via Zerotier One localhost:9993 API

### DISCUSSION THREAD

**Discussion by jerrychong25** [dec0dOS/zero-ui#109] [Q&A] [answered]
Hi all,

I would like to deauthorize and delete a member via Zerotier localhost:9993 API as mentioned in https://docs.zerotier.com/self-hosting/network-controllers/.

Unfortunately, the API unable to remove assgined IP (In this case **10.1.1.200**).

Steps To Reproduce The Error:
1. Have existing one online member ID in one network.
2. Call POST Deauthorize Member API
3. Call DELETE Member API
4. Call GET Member API
5. ipAssignments still have value of **10.1.1.200**

POST Deauthorize Member API Details:
```
Request:
curl -X POST "http://localhost:9993/controller/network/XXXXXXXXXXXXXXXXX/member/aaaaaaaaa" -H "X-ZT1-AUTH: auth_token_here" -d '{"authorized": false}'
```

DELETE Member API Details:
```
Request:
curl -X DELETE "http://localhost:9993/controller/network/XXXXXXXXXXXXXXXXX/member/aaaaaaaaa" -H "X-ZT1-AUTH: auth_token_here""

Response:
{}
```

GET Member API Details:
```
Request:
curl -X GET "http://localhost:9993/controller/network/XXXXXXXXXXXXXXXXX/member/aaaaaaaaa" -H "X-ZT1-AUTH: auth_token_here"

Response:
{
 "activeBridge": false,
 "address": "aaaaaaaaa",
 "authenticationExpiryTime": 0,
 "authorized": false,
 "capabilities": [],
 "creationTime": 1660278611038,
 "id": "563af8b3a5",
 "identity": "aaaaaaaaa:0:eacd8f65052054d842881f8bb530ea8e6750e262501783f2e8944ea16d4080684f25333dde47af796e27fac16bb8841494ecd164f5b0924b6beb1b0208699999",
 "ipAssignments": [
  "10.1.1.200"
 ],
 "lastAuthorizedCredential": null,
 "lastAuthorizedCredentialType": "api",
 "lastAuthorizedTime": 1660485003251,
 "lastDeauthorizedTime": 1661579147643,
 "noAutoAssignIps": false,
 "nwid": "XXXXXXXXXXXXXXXXX",
 "objtype": "member",
 "remoteTraceLevel": 0,
 "remoteTraceTarget": null,
 "revision": 8,
 "ssoExempt": false,
 "tags": [],
 "vMajor": 1,
 "vMinor": 10,
 "vProto": 12,
 "vRev": 1
}
```

After I've deleted a member, there is no way I could delete the assigned IP anymore.

Any solution to solve this issue via API only?

So that previously assigned IP could assign to new member ID.

Related:
- https://github.com/zerotier/ZeroTierOne/issues/859
- https://github.com/zerotier/ZeroTierOne/issues/258
- https://github.com/zerotier/ZeroTierOne/issues/1743

Thank you.

**Comment by dec0dOS** [comment]
Hello, @jerrychong25.

This issue is not related to the ZeroUI project, but the ZeroTier Controller itself.
ZeroTier DELETE HTTP method does not work as expected in some cases, so we have some workarounds on ZeroUI level.

**Reply by jerrychong25** [reply]
Hi @dec0dOS ,

Noted.

Will check this issue on ZeroTier Controller level.

Thanks for supporting.

**Comment by jerrychong25** [comment, marked as answer]
Hi @dec0dOS ,

Found the solution.

POST API can clear previously assgined IPs and deauthorize a member ID via following ways:
```
Request:
curl -X POST "http://localhost:9993/controller/network/XXXXXXXXXXXXXXX/member/aaaaaaaaaaaaa" -H "X-ZT1-AUTH: auth_token_here" -d '{"authorized": false, "ipAssignments": []}'
```

Just set **ipAssignments** as empty array will clear all previous assgined IPs.

Thanks for helping!
